### ISSUE TITLE
Add expected digest to finalize command

### ISSUE THREAD

**Issue by jmpesp** [oxidecomputer/omicron#3057]
Allow users importing a file to a disk via bulk writes to validate the sha256 of the contents of the disk after the upload.

**Comment by david-crespo** [comment]
So the idea is it would 400 if the provided digest didn’t match? Unfortunately the user has no recourse at that point but to start over, but I guess that’s fine as long as we communicate it.

**Comment by jmpesp** [comment]
Yeah - at that point, each bulk write would have completed with a 200, but if the digest doesn't match we know something else went wrong.
